bindings/packager/nodejs/src-ts/plugins/electron/index.ts

```diff
@@ -0,0 +1,211 @@
+import type { Config, Resource } from "../../config";
+import type { PackageJson } from "..";
+import fs from "fs-extra";
+import path from "path";
+import os from "os";
+import { download as downloadElectron } from "@electron/get";
+import extractZip from "extract-zip";
+import { Pruner, isModule, normalizePath } from "./prune";
+
+export default async function run(
+  appPath: string,
+  packageJson: PackageJson
+): Promise<Partial<Config> | null> {
+  let electronPath;
+  try {
+    electronPath = require.resolve("electron", {
+      paths: [appPath],
+    });
+  } catch (e) {
+    return null;
+  }
+
+  const userConfig = packageJson.packager || {};
+
+  const electronPackageJson = JSON.parse(
+    (
+      await fs.readFile(
+        path.resolve(path.dirname(electronPath), "package.json")
+      )
+    ).toString()
+  );
+
+  const zipPath = await downloadElectron(electronPackageJson.version);
+  const zipDir = await fs.mkdtemp(path.join(os.tmpdir(), ".packager-electron"));
+  await extractZip(zipPath, {
+    dir: zipDir,
+  });
+
+  const platformName = os.platform();
+  let resources: Resource[] = [];
+  let frameworks: string[] = [];
+  let debianFiles: {
+    [k: string]: string;
+  } | null = null;
+  let binaryPath;
+
+  const appTempPath = await fs.mkdtemp(
+    path.join(os.tmpdir(), packageJson.name || "app-temp")
+  );
+
+  const pruner = new Pruner(appPath, true);
+
+  const outDir = userConfig.outDir ? path.resolve(userConfig.outDir) : null;
+  const ignoredDirs = outDir && outDir !== process.cwd() ? [outDir] : [];
+
+  const filterFunc = (_name: string): boolean => true;
```

Comment:
this seems like it doesn't filter anything, why is it needed?

Reply:
it can be a user-defined filter in the future, I got most of it from https://github.com/electron/packager/blob/main/src/prune.js

Reply:
if it serves no purpose now, let's remove it and add it later if needed